deployment/ec2/amis.py

```diff
@@ -0,0 +1,40 @@
+"""Helper functions to handle EC2 related operations with Boto3"""
+
+import boto3
+import logging
+
+
+LOGGER = logging.getLogger(__file__)
+LOGGER.addHandler(logging.StreamHandler())
+LOGGER.setLevel(logging.INFO)
+
+
+def _prune_ami(ec2, ami):
+    """Actually deregister AMI and its associated snapshot"""
+    LOGGER.info('Identified that [%s] is eligible to be pruned..',
+                ami["ImageId"])
+
+    ec2.deregister_image(ImageId=ami["ImageId"])
```

Comment:
In Boto 2, this method had an option for deleting the associated root volume snapshot as well. It doesn't look like that's there any longer, but I still think we need to delete it, otherwise they'll accumulate and we won't have a great way to know they were associated with an old AMI.

You may need to use `describe_image_attribute` to get the `SnapshotId` via the `blockDeviceMapping`. Then, use that to call `delete_snapshot`.